src/psyclone/tests/dynamo0p3_basis_test.py

```diff
@@ -1438,6 +1438,7 @@ def test_eval_agglomerate(tmpdir):
 '''
 
 
+@pytest.mark.usefixtures("lfric_config")
```

Comment:
Ah, it turns out that it was due to the scope=module of the top fixture, this is quite dangerous because the config changes are preserved between tests and these happen in arbitrary order. Changing the scope to function has fixed all the issues I had here and I could remove this usefixtures.